variables.tf

```diff
@@ -259,3 +280,9 @@ variable "tags" {
   default     = null
   description = "(Optional) Tags of the resource."
 }
+
+variable "vnet_address_space" {
```

Comment:
THere should be a validation that this one cannot be filled in case the associated vnet it is filled otherwise you are going to chane the existing vnet.